sentry_sdk/tracing_utils.py

```diff
@@ -470,6 +470,46 @@ def from_incoming_header(cls, header):
 
         return Baggage(sentry_items, third_party_items, mutable)
 
+    @classmethod
+    def populate_from_transaction(cls, transaction):
+        # type: (Transaction) -> Baggage
+        """
+        Populate fresh baggage entry with sentry_items and make it immutable
+        if this is the head SDK which originates traces.
+        """
+
+        hub = transaction.hub or sentry_sdk.Hub.current
+        client = hub.client
+        sentry_items = {}  # type: Dict[str, str]
+
+        if not client:
+            return Baggage(sentry_items)
+
+        options = client.options or {}
+        user = (hub.scope and hub.scope._user) or {}
+
+        sentry_items["trace_id"] = transaction.trace_id
+
+        if options.get("environment"):
+            sentry_items["environment"] = options["environment"]
+
+        if options.get("release"):
+            sentry_items["release"] = options["release"]
+
+        if options.get("dsn"):
+            sentry_items["public_key"] = Dsn(options["dsn"]).public_key
+
+        if transaction.name and transaction.source != TRANSACTION_SOURCE_URL:
```

Comment:
Something like this would make the intention more readable:
```python
LOW_QUALITY_SOURCES = [TRANSACTION_SOURCE_URL, ]

if transaction.name and transaction.source not in LOW_QUALITY_SOURCES:
   ...
```